instrumentation/spring/spring-boot-autoconfigure-3/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/web/RestClientBeanPostProcessor.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure.instrumentation.web;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.spring.web.v3_1.SpringWebTelemetry;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.web.client.RestClient;
+
+public final class RestClientBeanPostProcessor implements BeanPostProcessor {
+
+  private final ObjectProvider<OpenTelemetry> openTelemetryProvider;
+
+  public RestClientBeanPostProcessor(ObjectProvider<OpenTelemetry> openTelemetryProvider) {
+    this.openTelemetryProvider = openTelemetryProvider;
+  }
+
+  @Override
+  public Object postProcessAfterInitialization(Object bean, String beanName) {
```

Comment:
A test verifying the instrumentation of a `RestClient` bean is missing?

Reply:
2 tests for 2 ways of using the rest client (see https://github.com/open-telemetry/opentelemetry.io/pull/4257/files#diff-ab82a329daf46a05ea680c4ff5dd748036189f16a08eb7882846e15ceadc9c28R531-R542)

- https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11038/files#diff-b33cbf72aac3d2048619a1f3a479a1ad6ff6f7d8e5e8da8829eaa6b9711b0c05
- https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11038/files#diff-e93c222a96e523196fa76738fd44fac8d05952a05eca3d54a55ad8734f995254R268-R270

Reply:
You test the `RestClient` bean post-processor: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/ff4842331dc89a40b5efa95c7a05952c662858b9/instrumentation/spring/spring-boot-autoconfigure-3/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/web/RestClientInstrumentationAutoConfigurationTest.java#L42

I would have preferred a test with a `RestClient` bean. But it's ok.

Reply:
This is also tested here: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/2a651f06f4a70c23b78b1e6d147c99243cd63f34/instrumentation/spring/spring-boot-autoconfigure-3/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/web/RestClientInstrumentationAutoConfigurationTest.java#L45-L59